examples/audio/sound_effects.rs

```diff
@@ -0,0 +1,207 @@
+    ///             GameSfx::SplashScreenJingle => "splash_screen_jingle",
+    ///             GameSfx::Victory => "victory",
+    ///             GameSfx::Defeat => "defeat",
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    fn play(&mut self, name: impl AsRef<str>, world: &mut World, settings: PlaybackSettings) {
+        let name = name.as_ref();
+        if let Some(sfx_list) = self.map.get_mut(name) {
+            // If you need precise control over the randomization order of your sound effects,
+            // store the RNG as a resource and modify these functions to take it as an argument.
+            let rng = &mut rand::thread_rng();
+
+            let index = rng.sample(Uniform::from(0..sfx_list.len()));
+            // We don't need a (slightly) more expensive strong handle here (which are used to keep assets loaded in memory)
+            // because a copy is always stored in the SoundEffects resource.
+            let source = sfx_list[index].clone_weak();
+
+            world.spawn(AudioBundle {
+                source,
+                // We want the sound effect to play once and then despawn.
+                settings,
+            });
+        } else {
+            warn!("Sound effect not found: {}", name);
+        }
+    }
+}
+
+impl FromWorld for SoundEffects {
+    // `FromWorld` is the simplest way to load assets into a resource,
+    // but you likely want to integrate this into your own asset loading strategy.
+    fn from_world(world: &mut World) -> Self {
+        let asset_server = world.get_resource::<AssetServer>().unwrap();
+        let mut map = HashMap::default();
+
+        // Load sound effects here.
+        // Using string parsing to strip numbered suffixes + `AssetServer::load_folder` is a good way to load many sound effects at once, but is not supported on Wasm or Android.
+        let button_press_sfxs = vec![
+            asset_server.load("sounds/button_press_1.ogg"),
+            asset_server.load("sounds/button_press_2.ogg"),
+            asset_server.load("sounds/button_press_3.ogg"),
+        ];
+        map.insert("button_press".to_string(), button_press_sfxs);
+
+        Self { map }
+    }
+}
+
+/// A custom command used to play sound effects.
+struct PlaySoundEffect {
+    name: String,
+    settings: PlaybackSettings,
+}
+
+impl Command for PlaySoundEffect {
+    fn apply(self, world: &mut World) {
+        // Access both the world and the resource we need from it using resource_scope
+        // which temporarily removes the SoundEffects resource from the world
+        world.resource_scope(|world, mut sound_effects: Mut<SoundEffects>| {
+            sound_effects.play(self.name, world, self.settings);
+        });
+    }
+}
+
+/// An "extension trait" used to make it convenient to play sound effects via [`Commands`].
+///
+/// This technique allows us to implement methods for types that we don't own,
+/// which can be used as long as the trait is in scope.
+trait SfxCommands {
```

Comment:
The rest of the structs use `SoundEffects` over `Sfx`. I have no preference for one or the other, but it would be nice to be consistent.

Reply:
Dropping my preference for `Sfx` here.

Reply:
After some back and forth in `bevy_quickstart`, I also prefer SFX, although it should not matter much for an example